pandas/io/parsers/python_parser.py

```diff
@@ -1228,14 +1243,16 @@ def detect_colspecs(
         return edge_pairs
 
     def __next__(self) -> list[str]:
+        # Argument 1 to "next" has incompatible type "Union[IO[str],
+        # ReadCsvBuffer[str]]"; expected "SupportsNext[str]"
         if self.buffer is not None:
             try:
                 line = next(self.buffer)
             except StopIteration:
                 self.buffer = None
-                line = next(self.f)
+                line = next(self.f)  # type: ignore[arg-type]
```

Comment:
@twoertwein Does ReadCsvBuffer suport __next__ in addition to __iter__?

Reply:
As far as I understand: As long as `.__iter__()` returns an object that has `__next__`, I don't think the file object itself has to have `__next__`.

Reply:
Hm weird, mypy complains about ``ReadCsvBuffer[str]``

Reply:
I'm sorry, you are right! I think only `read_fwf` calls next explicitly, or? `read_csv` does not explicitly call `next`. The following works with `read_csv`

```py
from io import StringIO

import pandas as pd


class Test:
    def __init__(self):
        self.buffer = StringIO("a,b,c\n0,1,2")

    @property
    def mode(self):
        return self.buffer.mode

    def fileno(self):
        return self.buffer.fileno()

    def seek(self, __offset, __whence=-1):
        if __whence != -1:
            return self.seek(__offset)
        return self.seek(__offset, __whence=__whence)

    def seekable(self):
        return self.buffer.seekable()

    def tell(self):
        return self.buffer.tell()

    def read(self, __n=None):
        self.buffer.read(__n)

    def __iter__(self):
        return self.buffer.__iter__()

    def readline(self):
        return self.buffer.readline()

    @property
    def closed(self):
        return self.buffer.closed()


print(pd.read_csv(Test(), engine="python"))
```

Reply:
Yep I think this is only called from read_fwf.

Can we add __next__ to ReadCsvBuffer without breaking anything (from a correctness standpoint)

Reply:
Would be fine with adding it or (probably better) creating `ReadFwfBuffer` which inherits from `ReadCsvBuffer` (not sure whether it needs everything that is in `ReadCsvBuffer`).

Also happy to have the ignores for now and address this in a later PR.

Reply:
Lets go with the ignore for now then until we have typed more. Can probably make a better decision then